python/ray/_private/worker.py

```diff
@@ -1902,6 +1902,11 @@ def connect(
         if mode == SCRIPT_MODE:
             raise e
         elif mode == WORKER_MODE:
+            if isinstance(e, grpc.RpcError) and e.code() in (
+                grpc.StatusCode.UNAVAILABLE,
+                grpc.StatusCode.UNKNOWN,
+            ):
+                raise e
```

Comment:
if `node.check_version_info` raise grpc.RpcError, it means GCS is unreachable, so the `ray._private.utils.publish_error_to_driver` will fail too, and we raise the Exception early can avoid wait to publish timeout, it will cost 60s for nothing